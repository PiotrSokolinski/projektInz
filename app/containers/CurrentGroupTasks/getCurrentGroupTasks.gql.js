import gql from 'graphql-tag'

export default gql`
  query getCurrentGroupTasks($id: ID!) {
    getGroupTasksToDoInProgress(id: $id) {
      id
      name
      status
    }
  }
`
