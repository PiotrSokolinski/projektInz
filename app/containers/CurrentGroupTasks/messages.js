/*
 * CurrentGroupTasks Messages
 *
 * This contains all the text for the CurrentGroupTasks container.
 */

import { defineMessages } from 'react-intl'

export const scope = 'app.containers.CurrentGroupTasks'

export default defineMessages({
  toDoTitle: {
    id: `${scope}.toDoTitle`,
    defaultMessage: 'To Do',
  },
  inProgressTitle: {
    id: `${scope}.inProgressTitle`,
    defaultMessage: 'In progress',
  },
})
