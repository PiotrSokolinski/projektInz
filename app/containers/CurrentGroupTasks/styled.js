import styled from 'styled-components'

import { Colors } from 'themes'

export const Container = styled.div`
  display: flex;
`

export const ToDoColumn = styled.div`
  flex: 1;
  border-right: 1px solid ${Colors.edward};
`

export const InProgressColumn = styled.div`
  flex: 1;
`

export const TitleColumn = styled.div`
  font-size: 15px;
  font-weight: bold;
  text-align: center;
  border-bottom: 1px solid ${Colors.edward};
`
