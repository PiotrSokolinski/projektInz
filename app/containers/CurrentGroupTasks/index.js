/**
 *
 * CurrentGroupTasks
 *
 */

import React from 'react'
import dayjs from 'dayjs'
import map from 'lodash/map'
// import PropTypes from 'prop-types'
import { FormattedMessage } from 'react-intl'

import TaskTile from 'components/TaskTile'

import messages from './messages'
import * as Styled from './styled'

const mockTasks = [
  {
    taskName: 'Task Name',
    taskDescription: `Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.`,
    author: {
      firstName: 'John',
      lastName: 'Doe',
      avatarUrl: '',
    },
    createdAt: dayjs().format('DD/MM/YYYY HH:mm:ss'),
    priority: 'High',
    status: 'Done',
    assignee: {
      firstName: 'John',
      lastName: 'Doe',
      avatarUrl: '',
    },
  },
  {
    taskName: 'Task Name',
    taskDescription: `Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.`,
    author: {
      firstName: 'John',
      lastName: 'Doe',
      avatarUrl: '',
    },
    createdAt: dayjs().format('DD/MM/YYYY HH:mm:ss'),
    priority: 'Medium',
    status: 'To Do',
    assignee: {
      firstName: 'John',
      lastName: 'Doe',
      avatarUrl: '',
    },
  },
  {
    taskName: 'Task Name',
    taskDescription: `Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.`,
    author: {
      firstName: 'John',
      lastName: 'Doe',
      avatarUrl: '',
    },
    createdAt: dayjs().format('DD/MM/YYYY HH:mm:ss'),
    priority: 'Low',
    status: 'Done',
    assignee: {
      firstName: 'John',
      lastName: 'Doe',
      avatarUrl: '',
    },
  },
]

const CurrentGroupTasks = () => (
  <Styled.Container>
    <Styled.ToDoColumn>
      <Styled.TitleColumn>
        <FormattedMessage {...messages.toDoTitle} />
      </Styled.TitleColumn>
      {map(mockTasks, (task, index) => (
        <TaskTile task={task} key={`el-to-do-${index}`} />
      ))}
    </Styled.ToDoColumn>
    <Styled.InProgressColumn>
      <Styled.TitleColumn>
        <FormattedMessage {...messages.inProgressTitle} />
      </Styled.TitleColumn>
      {map(mockTasks, (task, index) => (
        <TaskTile task={task} key={`el-in-progress-${index}`} />
      ))}
    </Styled.InProgressColumn>
  </Styled.Container>
)

CurrentGroupTasks.propTypes = {}

export default CurrentGroupTasks
