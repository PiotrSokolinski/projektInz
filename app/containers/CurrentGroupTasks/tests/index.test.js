/**
 *
 * Tests for CurrentGroupTasks
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from '@testing-library/react'
import { IntlProvider } from 'react-intl'

import CurrentGroupTasks from '../index'
import { DEFAULT_LOCALE } from '../../../i18n'

describe('<CurrentGroupTasks />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <CurrentGroupTasks />
      </IntlProvider>,
    )
    expect(spy).not.toHaveBeenCalled()
  })

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <CurrentGroupTasks />
      </IntlProvider>,
    )
    expect(firstChild).toMatchSnapshot()
  })
})
