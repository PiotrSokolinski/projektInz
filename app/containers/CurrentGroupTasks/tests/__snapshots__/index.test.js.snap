// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrentGroupTasks /> Should render and match the snapshot 1`] = `
<div
  class="styled__Container-sc-1vojkkb-0 ismdMZ"
>
  <div
    class="styled__ToDoColumn-sc-1vojkkb-1 kvchyu"
  >
    <div
      class="styled__TitleColumn-sc-1vojkkb-3 zZuWA"
    >
      <span>
        To Do
      </span>
    </div>
    <div
      class="styled__Container-sc-184074l-0 bVlVEA"
    >
      <div
        class="styled__Name-sc-184074l-1 kOpHSu"
      >
        Task Name
      </div>
      <div
        class="field styled__StatusSelect-sc-184074l-5 ddLUjA"
        id="statusSelect"
      >
        <div
          class="react-select css-2b097c-container"
        >
          <div
            class="select__control css-g95vz9-control"
          >
            <div
              class="select__value-container select__value-container--has-value css-1hwfws3"
            >
              <div
                class="select__single-value css-1uccc91-singleValue"
              >
                <div
                  class="styled__LabelContainer-sc-184074l-4 bNTAEt"
                >
                  <div
                    class="styled__Dot-sc-184074l-3 cRjzDn"
                    size="22"
                  />
                   To Do
                </div>
              </div>
              <input
                class="css-62g3xt-dummyInput"
                id="react-select-8-input"
                readonly=""
                tabindex="0"
                value=""
              />
            </div>
            <div
              class="select__indicators css-1wy0on6"
            />
          </div>
          <input
            name="status"
            type="hidden"
            value="To Do"
          />
        </div>
        
      </div>
      <svg
        aria-hidden="true"
        class="styled__TrashIcon-sc-184074l-2 jvDCxm StyledIconBase-sc-bdy9j4 jbPeIR"
        fill="currentColor"
        focusable="false"
        height="22"
        viewBox="0 0 24 24"
        width="22"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17.004 20l-.001-12h-10v12h10.001zm-4.001-10h2v8h-2v-8zm-4 0h2v8h-2v-8zm0-6h6v2h-6z"
          fill="none"
        />
        <path
          d="M5.003 20c0 1.103.897 2 2 2h10c1.103 0 2-.897 2-2V8h2V6h-4V4c0-1.103-.897-2-2-2h-6c-1.103 0-2 .897-2 2v2h-4v2h2v12zm4-16h6v2h-6V4zm-1 4h9l.001 12H7.003V8h1z"
        />
        <path
          d="M9.003 10h2v8h-2zm4 0h2v8h-2z"
        />
      </svg>
    </div>
    <div
      class="styled__Container-sc-184074l-0 bVlVEA"
    >
      <div
        class="styled__Name-sc-184074l-1 kOpHSu"
      >
        Task Name
      </div>
      <div
        class="field styled__StatusSelect-sc-184074l-5 ddLUjA"
        id="statusSelect"
      >
        <div
          class="react-select css-2b097c-container"
        >
          <div
            class="select__control css-g95vz9-control"
          >
            <div
              class="select__value-container select__value-container--has-value css-1hwfws3"
            >
              <div
                class="select__single-value css-1uccc91-singleValue"
              >
                <div
                  class="styled__LabelContainer-sc-184074l-4 bNTAEt"
                >
                  <div
                    class="styled__Dot-sc-184074l-3 cRjzDn"
                    size="22"
                  />
                   To Do
                </div>
              </div>
              <input
                class="css-62g3xt-dummyInput"
                id="react-select-9-input"
                readonly=""
                tabindex="0"
                value=""
              />
            </div>
            <div
              class="select__indicators css-1wy0on6"
            />
          </div>
          <input
            name="status"
            type="hidden"
            value="To Do"
          />
        </div>
        
      </div>
      <svg
        aria-hidden="true"
        class="styled__TrashIcon-sc-184074l-2 jvDCxm StyledIconBase-sc-bdy9j4 jbPeIR"
        fill="currentColor"
        focusable="false"
        height="22"
        viewBox="0 0 24 24"
        width="22"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17.004 20l-.001-12h-10v12h10.001zm-4.001-10h2v8h-2v-8zm-4 0h2v8h-2v-8zm0-6h6v2h-6z"
          fill="none"
        />
        <path
          d="M5.003 20c0 1.103.897 2 2 2h10c1.103 0 2-.897 2-2V8h2V6h-4V4c0-1.103-.897-2-2-2h-6c-1.103 0-2 .897-2 2v2h-4v2h2v12zm4-16h6v2h-6V4zm-1 4h9l.001 12H7.003V8h1z"
        />
        <path
          d="M9.003 10h2v8h-2zm4 0h2v8h-2z"
        />
      </svg>
    </div>
    <div
      class="styled__Container-sc-184074l-0 bVlVEA"
    >
      <div
        class="styled__Name-sc-184074l-1 kOpHSu"
      >
        Task Name
      </div>
      <div
        class="field styled__StatusSelect-sc-184074l-5 ddLUjA"
        id="statusSelect"
      >
        <div
          class="react-select css-2b097c-container"
        >
          <div
            class="select__control css-g95vz9-control"
          >
            <div
              class="select__value-container select__value-container--has-value css-1hwfws3"
            >
              <div
                class="select__single-value css-1uccc91-singleValue"
              >
                <div
                  class="styled__LabelContainer-sc-184074l-4 bNTAEt"
                >
                  <div
                    class="styled__Dot-sc-184074l-3 cRjzDn"
                    size="22"
                  />
                   To Do
                </div>
              </div>
              <input
                class="css-62g3xt-dummyInput"
                id="react-select-10-input"
                readonly=""
                tabindex="0"
                value=""
              />
            </div>
            <div
              class="select__indicators css-1wy0on6"
            />
          </div>
          <input
            name="status"
            type="hidden"
            value="To Do"
          />
        </div>
        
      </div>
      <svg
        aria-hidden="true"
        class="styled__TrashIcon-sc-184074l-2 jvDCxm StyledIconBase-sc-bdy9j4 jbPeIR"
        fill="currentColor"
        focusable="false"
        height="22"
        viewBox="0 0 24 24"
        width="22"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17.004 20l-.001-12h-10v12h10.001zm-4.001-10h2v8h-2v-8zm-4 0h2v8h-2v-8zm0-6h6v2h-6z"
          fill="none"
        />
        <path
          d="M5.003 20c0 1.103.897 2 2 2h10c1.103 0 2-.897 2-2V8h2V6h-4V4c0-1.103-.897-2-2-2h-6c-1.103 0-2 .897-2 2v2h-4v2h2v12zm4-16h6v2h-6V4zm-1 4h9l.001 12H7.003V8h1z"
        />
        <path
          d="M9.003 10h2v8h-2zm4 0h2v8h-2z"
        />
      </svg>
    </div>
  </div>
  <div
    class="styled__InProgressColumn-sc-1vojkkb-2 fqTFhY"
  >
    <div
      class="styled__TitleColumn-sc-1vojkkb-3 zZuWA"
    >
      <span>
        In progress
      </span>
    </div>
    <div
      class="styled__Container-sc-184074l-0 bVlVEA"
    >
      <div
        class="styled__Name-sc-184074l-1 kOpHSu"
      >
        Task Name
      </div>
      <div
        class="field styled__StatusSelect-sc-184074l-5 ddLUjA"
        id="statusSelect"
      >
        <div
          class="react-select css-2b097c-container"
        >
          <div
            class="select__control css-g95vz9-control"
          >
            <div
              class="select__value-container select__value-container--has-value css-1hwfws3"
            >
              <div
                class="select__single-value css-1uccc91-singleValue"
              >
                <div
                  class="styled__LabelContainer-sc-184074l-4 bNTAEt"
                >
                  <div
                    class="styled__Dot-sc-184074l-3 cRjzDn"
                    size="22"
                  />
                   To Do
                </div>
              </div>
              <input
                class="css-62g3xt-dummyInput"
                id="react-select-11-input"
                readonly=""
                tabindex="0"
                value=""
              />
            </div>
            <div
              class="select__indicators css-1wy0on6"
            />
          </div>
          <input
            name="status"
            type="hidden"
            value="To Do"
          />
        </div>
        
      </div>
      <svg
        aria-hidden="true"
        class="styled__TrashIcon-sc-184074l-2 jvDCxm StyledIconBase-sc-bdy9j4 jbPeIR"
        fill="currentColor"
        focusable="false"
        height="22"
        viewBox="0 0 24 24"
        width="22"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17.004 20l-.001-12h-10v12h10.001zm-4.001-10h2v8h-2v-8zm-4 0h2v8h-2v-8zm0-6h6v2h-6z"
          fill="none"
        />
        <path
          d="M5.003 20c0 1.103.897 2 2 2h10c1.103 0 2-.897 2-2V8h2V6h-4V4c0-1.103-.897-2-2-2h-6c-1.103 0-2 .897-2 2v2h-4v2h2v12zm4-16h6v2h-6V4zm-1 4h9l.001 12H7.003V8h1z"
        />
        <path
          d="M9.003 10h2v8h-2zm4 0h2v8h-2z"
        />
      </svg>
    </div>
    <div
      class="styled__Container-sc-184074l-0 bVlVEA"
    >
      <div
        class="styled__Name-sc-184074l-1 kOpHSu"
      >
        Task Name
      </div>
      <div
        class="field styled__StatusSelect-sc-184074l-5 ddLUjA"
        id="statusSelect"
      >
        <div
          class="react-select css-2b097c-container"
        >
          <div
            class="select__control css-g95vz9-control"
          >
            <div
              class="select__value-container select__value-container--has-value css-1hwfws3"
            >
              <div
                class="select__single-value css-1uccc91-singleValue"
              >
                <div
                  class="styled__LabelContainer-sc-184074l-4 bNTAEt"
                >
                  <div
                    class="styled__Dot-sc-184074l-3 cRjzDn"
                    size="22"
                  />
                   To Do
                </div>
              </div>
              <input
                class="css-62g3xt-dummyInput"
                id="react-select-12-input"
                readonly=""
                tabindex="0"
                value=""
              />
            </div>
            <div
              class="select__indicators css-1wy0on6"
            />
          </div>
          <input
            name="status"
            type="hidden"
            value="To Do"
          />
        </div>
        
      </div>
      <svg
        aria-hidden="true"
        class="styled__TrashIcon-sc-184074l-2 jvDCxm StyledIconBase-sc-bdy9j4 jbPeIR"
        fill="currentColor"
        focusable="false"
        height="22"
        viewBox="0 0 24 24"
        width="22"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17.004 20l-.001-12h-10v12h10.001zm-4.001-10h2v8h-2v-8zm-4 0h2v8h-2v-8zm0-6h6v2h-6z"
          fill="none"
        />
        <path
          d="M5.003 20c0 1.103.897 2 2 2h10c1.103 0 2-.897 2-2V8h2V6h-4V4c0-1.103-.897-2-2-2h-6c-1.103 0-2 .897-2 2v2h-4v2h2v12zm4-16h6v2h-6V4zm-1 4h9l.001 12H7.003V8h1z"
        />
        <path
          d="M9.003 10h2v8h-2zm4 0h2v8h-2z"
        />
      </svg>
    </div>
    <div
      class="styled__Container-sc-184074l-0 bVlVEA"
    >
      <div
        class="styled__Name-sc-184074l-1 kOpHSu"
      >
        Task Name
      </div>
      <div
        class="field styled__StatusSelect-sc-184074l-5 ddLUjA"
        id="statusSelect"
      >
        <div
          class="react-select css-2b097c-container"
        >
          <div
            class="select__control css-g95vz9-control"
          >
            <div
              class="select__value-container select__value-container--has-value css-1hwfws3"
            >
              <div
                class="select__single-value css-1uccc91-singleValue"
              >
                <div
                  class="styled__LabelContainer-sc-184074l-4 bNTAEt"
                >
                  <div
                    class="styled__Dot-sc-184074l-3 cRjzDn"
                    size="22"
                  />
                   To Do
                </div>
              </div>
              <input
                class="css-62g3xt-dummyInput"
                id="react-select-13-input"
                readonly=""
                tabindex="0"
                value=""
              />
            </div>
            <div
              class="select__indicators css-1wy0on6"
            />
          </div>
          <input
            name="status"
            type="hidden"
            value="To Do"
          />
        </div>
        
      </div>
      <svg
        aria-hidden="true"
        class="styled__TrashIcon-sc-184074l-2 jvDCxm StyledIconBase-sc-bdy9j4 jbPeIR"
        fill="currentColor"
        focusable="false"
        height="22"
        viewBox="0 0 24 24"
        width="22"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17.004 20l-.001-12h-10v12h10.001zm-4.001-10h2v8h-2v-8zm-4 0h2v8h-2v-8zm0-6h6v2h-6z"
          fill="none"
        />
        <path
          d="M5.003 20c0 1.103.897 2 2 2h10c1.103 0 2-.897 2-2V8h2V6h-4V4c0-1.103-.897-2-2-2h-6c-1.103 0-2 .897-2 2v2h-4v2h2v12zm4-16h6v2h-6V4zm-1 4h9l.001 12H7.003V8h1z"
        />
        <path
          d="M9.003 10h2v8h-2zm4 0h2v8h-2z"
        />
      </svg>
    </div>
  </div>
</div>
`;
